### PR TITLE
#325085: Add notification badge feature on Form to display on SimpleTab

### PR DESCRIPTION
**Pull request by mwagnerxaq**

### CODE REVIEW

File: eclipse-scout-core/src/status/Status.js [hunk 1]
```diff
@@ -93,7 +94,7 @@ export default class Status {
     if (!objects.equalsRecursive(this.children, o.children)) {
       return false;
     }
-    return objects.propertiesEquals(this, o, ['severity', 'message', 'invalidDate', 'invalidTime']);
+    return objects.propertiesEquals(this, o, ['severity', 'message', 'invalidDate', 'invalidTime', 'cssClassInternal']);
```
Comment (fschinkel) [inline eclipse-scout-core/src/status/Status.js]: do you mean _statusCssClass_?

File: eclipse-scout-core/src/tabbox/SimpleTab.js [hunk 2]
```diff
@@ -209,26 +209,35 @@ export default class SimpleTab extends Widget {
...
-          $statusIcon.addClass(status.cssClass());
+        if (status.iconId) {
+          let $statusIcon = this.$statusContainer.appendIcon(status.iconId, 'status');
+          if (status.cssClass()) {
+            $statusIcon.addClass(status.cssClass());
+          }
+          this._statusIconSpans.push($statusIcon);
+        }
+        if (status.message) {
+          let $statusMessage = this.$statusContainer.appendSpan('status message', status.message);
```
Comment (fschinkel) [inline eclipse-scout-core/src/tabbox/SimpleTab.js]: there is no css-class _message_ defined in SimpleTab.less

File: org.eclipse.scout.rt.client/src/main/java/org/eclipse/scout/rt/client/ui/form/AbstractForm.java [hunk 3]
```diff
@@ -906,6 +920,30 @@ private MultiStatus ensureMultiStatus(IStatus s) {
...
+  }
+
+  protected Optional<NotificationBadgeStatus> getNotificationBadgeStatus() {
+    final MultiStatus ms = getStatusInternal();
+    if (ms == null || !ms.containsStatus(NotificationBadgeStatus.class)) {
+      return Optional.empty();
+    }
+    return ms.getChildren().stream()
+        .filter(NotificationBadgeStatus.class::isInstance)
+        .map(status -> (NotificationBadgeStatus) status)
```
Comment (fschinkel) [inline org.eclipse.scout.rt.client/src/main/java/org/eclipse/scout/rt/client/ui/form/AbstractForm.java]: maybe _NotificationBadgeStatus.class::cast_?

File: org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/status/Status.java [hunk 4]
```diff
@@ -164,19 +176,14 @@ public boolean isMultiStatus() {
...
-      return (int) ((getOrder() - o.getOrder()));
-    }
-    else if (getMessage() != null && o.getMessage() != null) {
-      return getMessage().compareTo(o.getMessage());
-    }
-    else if (getMessage() != null) {
-      return -1;
-    }
-    return 0;
+    return Comparator
```
Comment (fschinkel) [inline org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/status/Status.java]: please format, spaces are missing

File: org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/status/Status.java [hunk 5]
```diff
@@ -147,6 +149,16 @@ public Status withCode(int code) {
     return this;
   }
 
+  @Override
+  public String getCssClass() {
+    return m_cssClass;
+  }
+
+  public Status withCssClass(String cssClass) {
```
Comment (fschinkel) [inline org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/status/Status.java]: please add a setter (like e.g. _severity_, _code_)

**Review by fschinkel — CHANGES_REQUESTED** [review]
Can you please add an example in the widgets app.

File: eclipse-scout-core/src/status/Status.js (hunk 1 above)
Comment (mwagnerxaq) [reply eclipse-scout-core/src/status/Status.js]: Done 👍

File: eclipse-scout-core/src/tabbox/SimpleTab.js (hunk 2 above)
Comment (mwagnerxaq) [reply eclipse-scout-core/src/tabbox/SimpleTab.js]: Done 👍

File: org.eclipse.scout.rt.client/src/main/java/org/eclipse/scout/rt/client/ui/form/AbstractForm.java (hunk 3 above)
Comment (mwagnerxaq) [reply org.eclipse.scout.rt.client/src/main/java/org/eclipse/scout/rt/client/ui/form/AbstractForm.java]: Done 👍

File: org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/status/Status.java (hunk 4 above)
Comment (mwagnerxaq) [reply org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/status/Status.java]: Done 👍

File: org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/status/Status.java (hunk 5 above)
Comment (mwagnerxaq) [reply org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/status/Status.java]: Done 👍

**Review by fschinkel — APPROVED** [review]

**Comment by fschinkel** [comment]
Jenkins run tests

**Pull request merged**